libethereum/Block.cpp

```diff
@@ -371,6 +371,7 @@ pair<TransactionReceipts, bool> Block::sync(BlockChain const& _bc, TransactionQu
 					else
 					{
 						clog(StateTrace) << t.sha3() << "Temporarily no gas left in current block (txs gas > block's gas limit)";
+						_tq.drop(t.sha3());
```

Comment:
I don't think we should drop the transaction here, especially not as part of the EIP150 change.